pkg/controller/node/nodecontroller.go

```diff
@@ -561,7 +561,9 @@ func (nc *NodeController) Run() {
 						} else {
 							glog.V(4).Info("Made sure that Node %v has no %v Taint", value.Value, oppositeTaint)
 						}
-						return true, 0
+
+						// remove the node from the queue after the adding/removal of taint is successful
+						return nc.zoneNotReadyOrUnreachableTainer[utilnode.GetZoneKey(node)].Remove(node.Name), 0
```

Comment:
It is not immediately obvious to me why the taint of the node at the time of addition to the map matters. It looks to me that the map stores only the name and UID of the node, not its taint. Looks like I need to read and understand the code better.